msom/src/main/java/pl/edu/uj/fais/wpz/msom/entities/Module.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package pl.edu.uj.fais.wpz.msom.entities;
+
+import java.util.HashSet;
+import java.util.Set;
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import pl.edu.uj.fais.wpz.msom.entities.abstracts.AbstractEntity;
+
+/**
+ * Basic Module (Single server) entity structure
+ * @author paweldylag
+ */
+@Entity
+public class Module extends AbstractEntity {
+
+    /** Module custom name */
+    private String name;
+    /** Number of servers cores */
+    private Integer cores;
+    /** Server's efficiency value */
+    private Integer efficiency;
+    /** Possible task types to process */
+    @ManyToMany(fetch = FetchType.EAGER, cascade = {CascadeType.ALL})
+    @JoinTable(name="task_types_per_module", 
+                joinColumns={@JoinColumn(name="module_id")}, 
+                inverseJoinColumns={@JoinColumn(name="tasktype_id")})
+    private Set<TaskType> taskTypes = new HashSet<>(0);
```

Comment:
W TaskTypeDao metoda removeTaskType nie uwzględnia tego, że moduły korzystają z typów zadań. Typ zadania wprawdzie nie zostanie usunięty, gdy moduł z niego korzysta, bo fajnie jest to tutaj rozwiązane na foreign key i hibernate na to nie zezwoli, ale wypadało by sprawdzać przed próbą usunięcia czy przypadkiem nie istnieją moduły korzystające z typu. Teraz zwracany jest wyjątek org.hibernate.exception.ConstraintViolationException: could not execute statement.

Swoją drogą wydzielił bym metody isUsedByAnyModules, isUsedByAnyTask do interfejsu TaskTypeDao, a metodę removeTaskType może zmienił na nadpisywaną metodę remove z interfejsu IDao<>. Byłoby bardziej intuicyjne korzystanie z metody remove (zamiast removeTaskType).

W testach jednostkowych dla TaskTypeDao też pasuje uzwględnić moduły przy próbie usunięcia.